### PR TITLE
update fal_securedownload for TYPO3 11

### PR DESCRIPTION
**Pull request by tritum**
update for compatibility with TYPO3 11

### CODE REVIEW

File: Classes/Configuration/ExtensionConfiguration.php [hunk 1]
```diff
@@ -45,7 +42,7 @@ private static function init()
     {
         if (!self::$isInitialized) {
             self::$isInitialized = true;
-            $extensionConfig = GeneralUtility::makeInstance(ExtensionConfigurationCore::class)->get('fal_securedownload');
+            $extensionConfig = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\TYPO3\CMS\Core\Configuration\ExtensionConfiguration::class)->get('fal_securedownload');
```
Comment (tritum) [inline Classes/Configuration/ExtensionConfiguration.php:45]: Just out of curiosity: why do we use the full namespace here and remove the use statement above?

File: Classes/EventListener/FolderChangedEventListener.php [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?php
-namespace BeechIt\FalSecuredownload\Hooks;
+namespace BeechIt\FalSecuredownload\EventListener;
```
Comment (tritum) [inline Classes/EventListener/FolderChangedEventListener.php:2]: Add copyright note.

File: Classes/EventListener/GeneratePublicUrlForResourceEventListener.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?php
+
```
Comment (tritum) [inline Classes/EventListener/GeneratePublicUrlForResourceEventListener.php:2]: Add copyright note.

File: Classes/EventListener/ModifyIconForResourcePropertiesEventListener.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+
```
Comment (tritum) [inline Classes/EventListener/ModifyIconForResourcePropertiesEventListener.php:2]: Add copyright note.

File: Classes/Events/AddCustomGroupsEvent.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?php
+
```
Comment (tritum) [inline Classes/Events/AddCustomGroupsEvent.php:2]: Add copyright note.

File: Classes/Events/BeforeFileDumpEvent.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
+<?php
+
```
Comment (tritum) [inline Classes/Events/BeforeFileDumpEvent.php:2]: Add copyright note.

File: Classes/Events/BeforeRedirectsEvent.php [hunk 7]
```diff
@@ -0,0 +1,65 @@
+<?php
+
```
Comment (tritum) [inline Classes/Events/BeforeRedirectsEvent.php:2]: Add copyright note.

File: Classes/EventListener/GeneratePublicUrlForResourceEventListener.php (hunk 3 above)
Comment (tritum) [reply Classes/EventListener/GeneratePublicUrlForResourceEventListener.php:2]: added

File: Classes/Events/AddCustomGroupsEvent.php (hunk 5 above)
Comment (tritum) [reply Classes/Events/AddCustomGroupsEvent.php:2]: added

File: Classes/Events/BeforeRedirectsEvent.php (hunk 7 above)
Comment (tritum) [reply Classes/Events/BeforeRedirectsEvent.php:2]: added

File: Classes/EventListener/FolderChangedEventListener.php (hunk 2 above)
Comment (tritum) [reply Classes/EventListener/FolderChangedEventListener.php:2]: was already there but updated it

File: Classes/EventListener/ModifyIconForResourcePropertiesEventListener.php (hunk 4 above)
Comment (tritum) [reply Classes/EventListener/ModifyIconForResourcePropertiesEventListener.php:2]: added

**Comment by joey-beech-it** [comment]
@tritum 

Thanks for the PR, we've planned some hours this week for testing and releasing.

**Comment by joey-beech-it** [comment]
@tritum A colleague of mine looked at the PR, he however concluded that it was not working yet. 

Could you give us a status update in terms of the state of this PR?
Should it be working out of the box or is there still work that has to be done?

**Comment by tritum** [comment]
@joey-beech-it Hi, i overdone it a little bit with the rector php 8 rules. So it kind of killed the php 7.4 compatibility.
No i reversed it so that it should be working with php 7.4 again.

**Comment by FamousWolf** [comment]
@joey-beech-it Looks like it's working correctly in both PHP 7.4 and 8.0. While testing I did run into some things not related to this update. Can we discuss that Monday?

**Comment by FamousWolf** [comment]
We'll probably make a new release at the end of this week.

**Pull request merged**